### PR TITLE
feat: add helper scripts for setting rtspauth secrets

### PR DESCRIPTION
**Pull request by ajcasagrande**
**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A**
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

**Secure**
```sh
cd edgex-compose/compose-builder
make clean
make run ds-usb-camera

# clone this PR
cd /path/to/device-usb-camera
bin/set-credentials.sh
# fill out values
```

**Non-Secure**

```sh
cd edgex-compose/compose-builder
make clean
make run no-secty ds-usb-camera

# clone this PR
cd /path/to/device-usb-camera
bin/set-credentials.sh
# fill out values
```

### CODE REVIEW

**Review by vyshali-chitikeshi — APPROVED** [review]
Validation looks good both in secure and non-secure modes.
able to set and/or edit rtsp credentials using set-credentials script both in secure and non-secure modes.

**Review by ChristianDarr-personal — APPROVED** [review]
LGTM

File: bin/utils.sh [hunk 1]
```diff
@@ -0,0 +1,348 @@
+#!/usr/bin/env bash
+
+#
+# Copyright (C) 2022-2023 Intel Corporation
```
Comment (presatish) [inline bin/utils.sh:4]:
```suggestion
# Copyright (C) 2023 Intel Corporation
```

File: bin/utils.sh [hunk 2]
```diff
@@ -0,0 +1,348 @@
...
+        if [ -z "${SECRET_PASSWORD}" ]; then
+            log_error "No password entered, exiting..."
+            return 1
+        fi
+    fi
+}
+
+# usage: try_set_argument "arg_name" "$@"
+# attempts to set the global variable "arg_name" to the next value from the command line.
+# if one is not provided, print error and return and error code.
```
Comment (presatish) [inline bin/utils.sh:168]:
```suggestion
# if one is not provided, print error and return an error code.
```

File: bin/utils.sh [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+}
+
+# create or update the insecure secrets by setting the 3 required fields in Consul
+set_insecure_secret() {
+    put_insecure_secrets_field "${SECRET_NAME}/SecretName"             "${SECRET_NAME}"
+    put_insecure_secrets_field "${SECRET_NAME}/SecretData/username"    "${SECRET_USERNAME}"
+    put_insecure_secrets_field "${SECRET_NAME}/SecretData/password"    "${SECRET_PASSWORD}"
+}
+
+# set the secure secrets by posting to the device service's secret endpoint
```
Comment (presatish) [inline bin/utils.sh:246]:
```suggestion
# set the secure secret by posting to the device service's secret endpoint
```
Is it not setting just a secret?

File: bin/utils.sh [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            \"value\":\"${SECRET_PASSWORD}\"
+        }
+    ]
+}"
+    do_curl "${payload}" \
+        -H "Authorization:Bearer ${REST_API_JWT}" \
+        -X POST "${DEVICE_SERVICE_URL}/api/v3/secret"
+}
+
+# helper function to set the secrets using either secure or insecure mode
```
Comment (presatish) [inline bin/utils.sh:271]:
```suggestion
# helper function to set the secret using either secure or insecure mode
```

**Comment by ajcasagrande** [comment]
Pending decision on https://github.com/edgexfoundry/go-mod-bootstrap/pull/542 . If that gets merged into 3.0, this PR will be closed as no longer needed.

**Comment by ajcasagrande** [comment]
Closing without merge, as https://github.com/edgexfoundry/go-mod-bootstrap/pull/542 has been approved and merged.

The new recommended way to set these values will be consistent across  secure and non-secure:

```bash
curl --data '{
    "apiVersion":"v3",
    "secretName": "rtspauth",
    "secretData":[
        {
            "key":"username",
            "value":"<pick-a-username>"
        },
        {
            "key":"password",
            "value":"<pick-a-secure-password>"
        }
    ]
}' -H Authorization:Bearer "<enter your JWT token here (make get-token)>" -X POST http://localhost:59983/api/v3/secret
```

(with the jwt only being required in secure mode)

**Pull request closed without merging**